### PR TITLE
Fix B-Cycle Date Handling

### LINKED ISSUE #1419: Update Austin B-Cycle Trips dataset
It looks like the last Checkout Date for this dataset is Nov 30, 2019. As per David Orr, we get their report by the 10th of each month, so it looks like Dec, Jan are missing. Mary Vo wanted to know when the dataset will be updated.

### PR DESCRIPTION
**Pull request by johnclary**
Fixes [#1419](https://github.com/cityofaustin/atd-data-tech/issues/1419).

### CODE REVIEW

**Comment by johnclary** [comment]
appreciate a review here. perhaps not my best work but trying not to get sucked into refactoring all this until focusing on airflow migration.

**Comment by johnclary** [comment]
btw, this ran successfully via atd-data-deploy on the server. b-cycle data is current: https://data.austintexas.gov/Transportation-and-Mobility/Austin-B-Cycle-Trips/tyfh-5r8s

File: transportation-data-publishing/bcycle/bcycle_trip_pub.py [hunk 1]
```diff
@@ -99,78 +66,66 @@ def handle_data(data):
...
+        except dropbox.exceptions.ApiError as e:
+
+            if "LookupError" in str(e):
+                # end loop when no file can be found
+                break
+            else:
+                raise e
+
+        data = handle_data(data)
+        data = format_checkout_date(data, next_file_date.year)
```
Comment (sergiogcx) [inline transportation-data-publishing/bcycle/bcycle_trip_pub.py:119]: Is the date data always provided? If not, shouldn't errors be handled in a try-catch statement here or within the function? Also, what would happen (for example) if the date is empty, would `format_checkout_date ` and socratautil be ok having an empty date?

**Review by sergiogcx — APPROVED** [review]
Looks good! Ship soda! 🚢  

soda

**Review by mateoclarke — APPROVED** [review]
🚲 💨

File: transportation-data-publishing/bcycle/bcycle_trip_pub.py (hunk 1 above)
Comment (johnclary) [reply transportation-data-publishing/bcycle/bcycle_trip_pub.py:119]: @sergiogcx yes, appreciate that concern. these CSV's are created by B-Cycle staff and uploaded to dropbox, and the entire ETL is dependent on that CSV being formatted exactly according to the format we've agreed upon. so this whole process is very brittle. 

to answer your question: if the date is empty, the script should and will fail. every record must have a checkout date.

**Comment by johnclary** [comment]
thanks @mateoclarke @sergiogcx !

**Pull request merged**